File: dpo_train/translation_output/java-py/HumanEval_57_monotonic.py
from typing import List

def monotonic(l: List[int]) -> bool:
    if len(l) == 0:
        return True
    if len(l) == 1:
        return monotonic(l[0])
    return monotonic(l[0:1]) and monotonic(l[1:])

def check(candidate):
    assert candidate([1, 2, 4, 10]) == True
    assert candidate([1, 2, 4, 20]) == True
    assert candidate([1, 20, 4, 10]) == False
    assert candidate([4, 1, 0, -10]) == True
    assert candidate([4, 1, 1, 0]) == True
    assert candidate([1, 2, 3, 2, 5, 60]) == False
    assert candidate([1, 2, 3, 4, 5, 60]) == True
    assert candidate([9, 9, 9, 9]) == True

def test_check():
    check(monotonic)

test_check()

File: codes/src-tgt/py-cpp/HumanEval_57_monotonic.cpp
### Python version

from typing import List

def monotonic(l: List[int]) -> bool:
    if l == sorted(l) or l == sorted(l, reverse=True):
        return True
    return False

### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
bool monotonic(std::vector<long> l) {
    return std::is_sorted(l.begin(), l.end()) || std::is_sorted(l.rbegin(), l.rend());
}

